pyuvdata/uvdata/ms.py

```diff
@@ -126,11 +886,715 @@ def find_not_ascii(s):
+                application.append("pyuvdata")
+                message.append(line)
+                obj_id.append(0)
+                obs_id.append(-1)
+                origin.append("pyuvdata")
+                priority.append("INFO")
+                times.append(Time.now().mjd * 3600.0 * 24.0)
+
+        history_table = tables.table(filepath + "::HISTORY", ack=False, readonly=False)
+
+        nrows = len(message)
+        history_table.addrows(nrows)
+
+        # the first two lines below break on python-casacore < 3.1.0
+        history_table.putcol("APP_PARAMS", np.asarray(app_params)[:, np.newaxis])
+        history_table.putcol("CLI_COMMAND", np.asarray(cli_command)[:, np.newaxis])
+        history_table.putcol("APPLICATION", application)
+        history_table.putcol("MESSAGE", message)
+        history_table.putcol("OBJECT_ID", obj_id)
+        history_table.putcol("OBSERVATION_ID", obs_id)
+        history_table.putcol("ORIGIN", origin)
+        history_table.putcol("PRIORITY", priority)
+        history_table.putcol("TIME", times)
+
+        history_table.done()
+
+    def write_ms(
+        self,
+        filepath,
+        force_phase=False,
+        clobber=False,
+        run_check=True,
+        check_extra=True,
+        run_check_acceptability=True,
+        strict_uvw_antpos_check=False,
+    ):
+        """
+        Write a CASA measurement set (MS).
+
+        Parameters
+        ----------
+        filepath : str
+            The MS file path to write to.
+        force_phase : bool
+            Option to automatically phase drift scan data to zenith of the first
+            timestamp.
+        clobber : bool
+            Option to overwrite the file if it already exists.
+        run_check : bool
+            Option to check for the existence and proper shapes of parameters
+            before writing the file.
+        check_extra : bool
+            Option to check optional parameters as well as required ones.
+        run_check_acceptability : bool
+            Option to check acceptable range of the values of parameters before
+            writing the file.
+        strict_uvw_antpos_check : bool
+            Option to raise an error rather than a warning if the check that
+            uvws match antenna positions does not pass.
+
+        """
+        if not casa_present:  # pragma: no cover
+            raise ImportError(no_casa_message) from casa_error
+
+        if run_check:
+            self.check(
+                check_extra=check_extra,
+                run_check_acceptability=run_check_acceptability,
+                strict_uvw_antpos_check=strict_uvw_antpos_check,
+            )
+
+        if os.path.exists(filepath):
+            if clobber:
+                print("File exists; clobbering")
+            else:
+                raise IOError("File exists; skipping")
+
+        # TODO: the phasing requirement can probably go away once we have proper
+        # multi-object support.
```

Comment:
What are we missing on this front?

Reply:
CASA does not appear to support "unphased" data, so we'll still need to keep this check (I actually need to add a check for when there is unphased data in a multi-obj data set).

Reply:
Ok, that sounds reasonable. Can we update the comment?

Reply:
Yup, will do.

Reply:
Comment updated in 3068e067.